src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -495,4 +500,135 @@ public void close() throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
+        if (inspectionPortEntity == null) {
+            String msg = String.format("inspection port element does not exists id: %s\n",
+                    inspectionPort.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            inspectionPortEntity.setEgressPort((NetworkElementEntity) inspectionPort.getEgressPort());
+            inspectionPortEntity.setIngressPort((NetworkElementEntity) inspectionPort.getIngressPort());
+
+            return this.em.merge(inspectionPortEntity);
+        });
+    }
+
+    public List<String> getInspectionPortsIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+
+            List<? extends InspectionPortEntity> elements = this.utils.txInspectionPortEntities();
+            List<String> elementList = new ArrayList<String>();
+
+            if (!elements.isEmpty()) {
+                elementList = elements.stream().map(InspectionPortEntity::getElementId).collect(Collectors.toList());
+            }
+
+            return elementList;
+        });
+    }
+
+    public NetworkElement registerPort(NetworkElement nwElement) throws Exception {
+
+        NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(nwElement.getElementId());
+        if (element != null) {
+            String msg = String.format("Network element already exists id: %s\n", nwElement.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            NetworkElementEntity portElement = new NetworkElementEntity();
+            portElement.setElementId(nwElement.getElementId());
+            portElement.setMacAddresses(nwElement.getMacAddresses());
+            portElement.setPortIPs(nwElement.getPortIPs());
+            portElement.setDeviceOwnerId(((NetworkElementEntity) nwElement).getDeviceOwnerId());
+            portElement.setParentId(nwElement.getParentId());
+
+            this.em.merge(portElement);
+
+            return portElement;
+        });
+    }
+
+    public NetworkElement updatePort(NetworkElement nwElement) throws Exception {
+
+        NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(nwElement.getElementId());
+        if (element == null) {
+            String msg = String.format("Cannot find the network element id: %s\n", nwElement.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            element.setMacAddresses(nwElement.getMacAddresses());
+            element.setPortIPs(nwElement.getPortIPs());
+            element.setDeviceOwnerId(((NetworkElementEntity) nwElement).getDeviceOwnerId());
```

Comment:
this is an unsafe cast. How are we guaranteed to receive NetworkElementEntity? maybe just modify the method to take a NetworkElementEntity